android/src/main/kotlin/com/github/fluttercommunity/flutterdownloader/FlutterDownloaderPlugin.kt

```diff
@@ -0,0 +1,102 @@
+package com.github.fluttercommunity.flutterdownloader
+
+import android.util.Log
+import androidx.work.*
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.BinaryMessenger
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+
+/** FlutterDownloaderPlugin */
+class FlutterDownloaderPlugin : FlutterPlugin, MethodCallHandler {
+    companion object {
+        private const val TAG = "FlutterDownloaderPlugin"
+        private const val channelId = "fluttercommunity/flutter_downloader"
+        private lateinit var workManager: WorkManager
+        internal var messenger: BinaryMessenger? = null
+
+        fun getBackChannel(urlHash: String) = MethodChannel(requireNotNull(messenger), "$channelId/$urlHash")
+        val tempDir get() = tempPath
+        private lateinit var tempPath: String
+        private var dartUserAgent: String? = null
+    }
+    private var channel: MethodChannel? = null
+
+    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        messenger = flutterPluginBinding.binaryMessenger
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, channelId)
+        channel?.setMethodCallHandler(this)
+        workManager = WorkManager.getInstance(flutterPluginBinding.applicationContext)
+        tempPath = flutterPluginBinding.applicationContext.cacheDir.absolutePath
+    }
+
+    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        channel?.setMethodCallHandler(null)
+        channel = null
+        messenger = null
+    }
+
+    override fun onMethodCall(call: MethodCall, result: Result) {
+        when (call.method) {
+            "getCacheDir" -> result.success(tempPath)
+            "resume" -> resume(call, result)
+            "pause" -> pause(call, result)
+            else -> result.notImplemented()
+        }
+    }
+
+    /**
+     * Resume or start the download
+     *
+     * Returns true if successful, but will emit a status update that the background task is running
```

Comment:
```suggestion
     * Resumes the existing download, or starts a new download.
     *
     * Returns true if successful, but will emit a status update that the background task is running
```

Also, I think it'd be nice if this method explicitly specified what argument it needs. Maybe we could do some pre-validation in `onMethodCall(call, result)`, extract the arguments from `call` there, and run `resume()` with these args?